File: webapp/src/main/vueapp/src/logic/models/Locality.ts
import {GenericEntity} from '~/logic/models/utils/GenericEntity';

export class Locality extends GenericEntity<Locality> {
    private _id: number;
    private _name: string;
    private _provinceId: number;

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get name(): string {
        return this._name;
    }

    public set name(value: string) {
        this._name = value;
    }

    public get provinceId(): number {
        return this._provinceId;
    }

    public set provinceId(value: number) {
        this._provinceId = value;
    }
}

File: models/src/main/java/ar/edu/itba/paw/models/Locality.java
package ar.edu.itba.paw.models;

import javax.persistence.*;

@Entity
@Table(
        name = "system_locality",
        indexes = {
                @Index(columnList = "locality_id", name = "system_locality_locality_id_uindex", unique = true),
        }
)
public class Locality extends GenericModel<Integer> {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "system_locality_locality_id_seq")
    @SequenceGenerator(sequenceName = "system_locality_locality_id_seq", name = "system_locality_locality_id_seq", allocationSize = 1)
    @Column(name = "locality_id")
    private Integer id;
    @Column(name = "name", nullable = false)
    private String name;
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "province_id", nullable = false)
    private Province province;

    @Override
    public Integer getId() {
        return this.id;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Province getProvince() {
        return this.province;
    }

    public void setProvince(Province province) {
        this.province = province;
    }

    @Override
    protected boolean isSameType(Object o) {
        return o instanceof Locality;
    }
}
